src/group_theory/complement.lean

```diff
@@ -76,16 +76,74 @@ end
 @[to_additive] lemma is_complement'_comm : is_complement' H K ↔ is_complement' K H :=
 ⟨is_complement'.symm, is_complement'.symm⟩
 
-@[to_additive] lemma is_complement_top_singleton {g : G} :
-  is_complement (⊤ : set G) {g} :=
+@[to_additive] lemma is_complement_top_singleton {g : G} : is_complement (⊤ : set G) {g} :=
 ⟨λ ⟨x, _, rfl⟩ ⟨y, _, rfl⟩ h, prod.ext (subtype.ext (mul_right_cancel h)) rfl,
   λ x, ⟨⟨⟨x * g⁻¹, ⟨⟩⟩, g, rfl⟩, inv_mul_cancel_right x g⟩⟩
 
-@[to_additive] lemma is_complement_singleton_top {g : G} :
-  is_complement ({g} : set G) (⊤ : set G) :=
+@[to_additive] lemma is_complement_singleton_top {g : G} : is_complement ({g} : set G) ⊤ :=
 ⟨λ ⟨⟨_, rfl⟩, x⟩ ⟨⟨_, rfl⟩, y⟩ h, prod.ext rfl (subtype.ext (mul_left_cancel h)),
   λ x, ⟨⟨⟨g, rfl⟩, g⁻¹ * x, ⟨⟩⟩, mul_inv_cancel_left g x⟩⟩
 
+@[to_additive] lemma is_complement_singleton_left {g : G} : is_complement {g} S ↔ S = ⊤ :=
+begin
+  refine ⟨λ h, top_le_iff.mp (λ x hx, _), λ h, (congr_arg _ h).mpr is_complement_singleton_top⟩,
+  obtain ⟨y, hy⟩ := h.2 (g * x),
+  conv_rhs at hy { rw ← (show y.1.1 = g, from y.1.2) },
+  rw ← mul_left_cancel hy,
+  exact y.2.2,
```

Comment:
I'm not sure this is any better but you can avoid `conv` at the cost of a heavier `obtain`:
```suggestion
  obtain ⟨⟨⟨z, (hz : z = g)⟩, ⟨y, hy'⟩⟩, (hy : z * y = g * x)⟩ := h.2 (g * x),
  rw hz at hy,
  rw ← mul_left_cancel hy,
  exact hy',
```

Reply:
Thanks. I managed to golf it a bit further.